docs/getting_started/typescript/hello_world_in_typescript/index.md

```diff
@@ -88,142 +158,155 @@ Create the `package.json` file in the root of your project and add the following
 
-Create the file `activities.ts` in the `src` directory:
+You also set the Retry Policy for Activities this way. In this example, you're using the default Retry Policy values, but by leaving it in you have a more clear picture of what happens. Note that the `maximumAttempts` is commented out, which means there's no limit to the number of times Temporal will retry your Activities if they fail.
 
-Add the following code to define a `greet` function:
+With the imports and options in place, you can define the Workflow itself. In the TypeScript SDK, you implement a Workflow the same way you define an Activity; using an exportable async TypeScript function. Add the following code to call both Activities, using the value of the first as the input to the second:
 
-<!--SNIPSTART typescript-hello-activity-->
-[hello-world/src/activities.ts](https://github.com/temporalio/samples-typescript/blob/main/hello-world/src/activities.ts)
 ```ts
-export async function greet(name: string): Promise<string> {
-  return `Hello, ${name}!`;
+// The Temporal Workflow.
+export async function getAddressFromIP(name: string): Promise<string> {
+
+  try {
+    const ip = await getIP();
+    try {
+      const location = await getLocationInfo(ip);
+      return `Hello, ${name}. Your IP is ${ip} and your location is ${location}`;
+    } catch (e) {
+      throw new workflow.ApplicationFailure("Failed to get location");
+    }
+  } catch (e) {
+    throw new workflow.ApplicationFailure("Failed to get IP");
+  }
+
 }
 ```
-<!--SNIPEND-->
 
-You've completed the logic for the application; you have a Workflow and an Activity defined. Next, you'll write code to configure and launch a Worker.
+This code uses a `try/catch` block, but because you've configured unlimited retries, there won't be any exceptions caught. However, if you change the Retry Policy's maximum retries, or you specify specific non-retryable exceptions, this code will be in place to handle those errors.
+
+Next you'll create a Worker that executes the Workflow and Activities.
 
 ## Configure and run a Worker
 
-A [Worker](https://docs.temporal.io/concepts/what-is-a-worker) hosts Workflow and Activity functions and executes them one at a time. The Temporal Server tells the Worker to execute a specific function from information it pulls from the [Task Queue](https://docs.temporal.io/concepts/what-is-a-task-queue). After the Worker runs the code, it communicates the results back to the Temporal Server.
+When you start a Temporal Workflow, the Workflow and its Activities get scheduled on the Temporal Service's [Task Queue](https://docs.temporal.io/concepts/what-is-a-task-queue).  A [Worker](https://docs.temporal.io/concepts/what-is-a-worker) hosts Workflow and Activity functions and polls the Task Queue for tasks related to those Workflows and Activities. After the Worker runs the code, it communicates the results back to the Temporal Service where they're stored in the Event History.
```

Comment:
I would add a sentence to this to explain WHY the results are stored in the Event History. Consider adding something like:

```suggestion
When you start a Temporal Workflow, the Workflow and its Activities get scheduled on the Temporal Service's [Task Queue](https://docs.temporal.io/concepts/what-is-a-task-queue). A [Worker](https://docs.temporal.io/concepts/what-is-a-worker) hosts Workflow and Activity functions and polls the Task Queue for tasks related to those Workflows and Activities. After the Worker runs the code, it communicates the results back to the Temporal Service where they're stored in the Event History. This records the Workflow's entire execution, enabling features like fault tolerance by allowing the Workflow to replay in case of Worker crashes or restarts.
```